sdk/core/core/inc/az_http_policy.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_POLICY_H
+#define AZ_HTTP_POLICY_H
+
+#include <az_contract.h>
+#include <az_http_request_builder.h>
+#include <az_pair.h>
+#include <az_result.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <_az_cfg_prefix.h>
+
+// PipelinePolicies
+//   Policies are non-allocating caveat the TransportPolicy
+//   Transport policies can only allocate if the transport layer they call allocates
+// Client ->
+//  ===HttpPipelinePolicies===
+//    UniqueRequestID
+//    Retry
+//    Authentication
+//    Logging
+//    Buffer Response
+//    Distributed Tracing
+//    TransportPolicy
+//  ===Transport Layer===
+
+typedef struct az_http_policy az_http_policy;
```

Comment:
Move the struct defined below in this file, and declare it as typedef /*no name*/ { ... } struct_name;

Reply:
Unfortunately the named struct has to be defined this way because the struct contains a function pointer which has as an argument the struct.